go/ct/st/block_context.go

```diff
@@ -32,6 +34,44 @@ type BlockContext struct {
 	TimeStamp   uint64     // Block's timestamp in unix time in seconds
 }
 
+// NewBlockContext creates a new block context
+func NewBlockContext() *BlockContext {
+	return &BlockContext{BlobHashes: []vm.Hash{}}
+}
+
+func (b *BlockContext) Clone() *BlockContext {
+	clone := *b
+	clone.BlobHashes = slices.Clone(b.BlobHashes)
+	return &clone
+}
+
+func (b *BlockContext) Equal(other *BlockContext) bool {
+	isEqual := true
+
+	isEqual = isEqual && b.BaseFee.Eq(other.BaseFee)
+	isEqual = isEqual && b.BlobBaseFee.Eq(other.BlobBaseFee)
+	isEqual = isEqual && (b.BlockNumber == other.BlockNumber)
+	isEqual = isEqual && b.ChainID.Eq(other.ChainID)
+	isEqual = isEqual && (b.CoinBase == other.CoinBase)
+	isEqual = isEqual && (b.GasLimit == other.GasLimit)
+	isEqual = isEqual && b.GasPrice.Eq(other.GasPrice)
+	isEqual = isEqual && (b.PrevRandao == other.PrevRandao)
+	isEqual = isEqual && (b.TimeStamp == other.TimeStamp)
+
+	if len(b.BlobHashes) != len(other.BlobHashes) {
+		isEqual = false
+	} else {
+		for i, hash := range b.BlobHashes {
+			if !slices.Equal(hash[:], other.BlobHashes[i][:]) {
+				isEqual = false
+				break
+			}
+		}
+	}
```

Comment:
This should be the same as
```suggestion
	isEqual = isEqual && (slices.Equal(b.BlobHashes, other.BlobHashes))
```